tests/libwnbd_tests/mock_wnbd_daemon.cc

```diff
@@ -234,6 +237,85 @@ void MockWnbdDaemon::Unmap(
         handler->MockStatus, NULL, 0);
 }
 
+void MockWnbdDaemon::PersistentReserveIn(
+    PWNBD_DISK Disk,
+    UINT64 RequestHandle,
+    UINT8 ServiceAction)
+{
+    MockWnbdDaemon* handler = nullptr;
+    ASSERT_FALSE(WnbdGetUserContext(Disk, (PVOID*)&handler));
+
+    WnbdRequestType RequestType = WnbdReqTypePersistResIn;
+    WNBD_IO_REQUEST WnbdReq = { 0 };
+    WnbdReq.RequestType = RequestType;
+    WnbdReq.RequestHandle = RequestHandle;
+    WnbdReq.Cmd.PersistResIn.ServiceAction = ServiceAction;
+
+    handler->ReqLog.AddEntry(WnbdReq);
+
+    switch (ServiceAction)
+    {
+    case RESERVATION_ACTION_READ_KEYS:
+    case RESERVATION_ACTION_READ_RESERVATIONS:
+        break;
+    default:
+        WnbdSetSense(
+            &handler->MockStatus,
+            SCSI_SENSE_ILLEGAL_REQUEST,
+            SCSI_ADSENSE_ILLEGAL_COMMAND);
+        break;
+    }
+
+    handler->SendIoResponse(
+        RequestHandle, RequestType,
+        handler->MockStatus, NULL, 0);
```

Comment:
So we have a "persistent reservation in" command that doesn't actually return any data. It's ok for now, but we'll eventually have to ensure that persistent reservation data buffers get transferred accordingly.